R-packages/evalcast/vignettes/intro-evalcast.Rmd

```diff
@@ -44,87 +48,142 @@ library(evalcast)
+
+Let's look one location:
+
 ```{r}
-pc <- predictions_cards_cmu[[1]]
-ca_fips = covidcast::name_to_fips("California")
-ca_code = substr(ca_fips, 0, 2) # States are identified by a 2 character code
-pc[pc$location == ca_code, ]$forecast_distribution[[1]]
+predictions_cards_cmu %>% 
+  filter(geo_value == "mi", forecast_date == forecast_dates_july[1]) %>%
+  select(geo_value, quantile, value, forecast_date) %>% str()
 ```
 
+Forecasts are generally probabilistic, meaning that rather than predicting a single value, a distribution of values at different quantiles. For example, the Covidhub ensemble forecaster would have predicted the following distribution in California for `r predictions_cards_cmu$ahead[1]` week(s) after `r predictions_cards_cmu$forecast_date[1]`
+```{r}
+predictions_cards_ens %>%
+  filter(geo_value == "ca", forecast_date == forecast_dates_july[1]) %>%
+  select(quantile, value) %>% str()
+```
+
+An `NA` in the `quantile` column indicates a point forecast (possibly different from the median).
+
+
 ### Evaluate the performance of a forecaster
 
 Now that we've made these predictions, we'd like to know how well they perform.  By default, `evalcast` uses the following three performance measures, but these can be easily substituted:
 
 ```{r}
 err_measures <- list(wis = weighted_interval_score,
                      ae = absolute_error,
-                     coverage_80 = interval_coverage(alpha = 0.2))
+                     coverage_80 = interval_coverage(coverage = 0.8))
 ```
 
 We may now create scorecards for each forecaster's predictions or for a filtered subset. 
 
 ```{r}
-scorecards_yyg <- evaluate_predictions(filter_predictions(predictions_cards_yyg, ahead = 3), 
-                                   err_measures, 
-                                   backfill_buffer = 10)
-pcards_cmu_death3 <- filter_predictions(predictions_cards_cmu, 
-                                        ahead = 3,
-                                        response_data_source = "jhu-csse",
-                                        response_signal = "deaths_incidence_num")
-scorecards_cmu <- evaluate_predictions(pcards_cmu_death3, 
-                                   err_measures, 
-                                   backfill_buffer = 10)
-scorecards <- evaluate_predictions(filter_predictions(predictions_cards, ahead = 3), 
-                                   err_measures, 
-                                   backfill_buffer = 10)
-```
-
-Let's look at the `baseline_forecaster`'s scorecard for 3-epiweek ahead forecasts.
+scorecards <- evaluate_predictions(
+  bind_rows(predictions_cards, 
+            predictions_cards_cmu, 
+            predictions_cards_ens) %>%
+    # not always data for these locations
```

Comment:
Does this cause an issue somewhere if we don't remove them? If so, is there a way to have the functions automatically identify / handle that case?

Reply:
`intersect_averagers()` should take care of it. Though in keeping with our discussion yesterday, I think it's good to emphasize that one should be aware of the issue when performing these analyses.

Reply:
I removed the line for now, since I thought it was more confusing than illuminating. It didn't describe why it's important or how to determine which locations to remove. It is probably a good idea to build in `intersect_averagers` into the plotting functions to prevent users from analyzing non-comparable data.